### DISCUSSION TITLE
Build Schema from JSON introspection query

### DISCUSSION THREAD

**Discussion by ruudk** [webonyx/graphql-php#850] [Q&A] [answered]
Is it possible to build a schema from a JSON introspection query using https://webonyx.github.io/graphql-php/schema-definition-language/?

This doesn't work:
```php
<?php
use GraphQL\Utils\BuildSchema;

$contents = file_get_contents('schema.json');
$schema = BuildSchema::build($contents);
```
```
PHP Fatal error:  Uncaught Syntax Error: Expected Name, found String "data"

GraphQL request (1:2)
1: {"data":{"__schema":{"queryType":{"name":"Query"}...
```

Why? I want to parse the schema of an external GraphQL backend using this library.

**Comment by spawnia** [comment, marked as answer]
```php
\GraphQL\Utils\BuildClientSchema::build(
    json_decode(file_get_contents('schema.json'), true)
);
```

**Reply by ruudk** [reply]
Thank you 🙌 , this worked:
```php
\GraphQL\Utils\BuildClientSchema::build(
    json_decode(file_get_contents('schema.json'), true)['data']
);
```

**Reply by spawnia** [reply]
That is assuming you are directly saving the result of calling the introspection query to `schema.json`. I would recommend validating if it contains `errors` first, and only then picking `data` from it and save that to the file.

**Reply by ruudk** [reply]
Will do, thank you @spawnia
